src/extension/src/scripts/generate-test.ts

```diff
@@ -0,0 +1,129 @@
+import { CoreTemplateStudio } from "../coreTemplateStudio";
+import { CONSTANTS } from "../CONSTANTS";
+
+let instance: CoreTemplateStudio;
+let backend: string[] = [];
+let frontend: string[] = [];
+let pagesObj: any[] = [];
+const projType = "FullStackWebApp";
+let syncAttemptNum = 0;
+
+function delay(time: number) {
+  return new Promise(function(resolve) {
+    setTimeout(resolve, time);
+  });
+}
+
+let attemptSync: any = (
+  instanceObj: CoreTemplateStudio,
+  syncAttemptNum: number
+) => {
+  if (syncAttemptNum < CONSTANTS.API.MAX_SYNC_REQUEST_ATTEMPTS) {
+    return instanceObj
+      .sync({
+        port: instance.getPort(),
+        payload: { path: CONSTANTS.API.DEVELOPMENT_PATH_TO_TEMPLATES },
+        liveMessageHandler: value => {
+          value;
+        }
+      })
+      .then(() => {
+        instance
+          .getFrameworks(projType)
+          .then(frameworks => {
+            frameworks.forEach(
+              (obj: { tags: { type: string }; name: string }) => {
+                if (obj.tags.type == "frontend") {
+                  frontend.push(obj.name);
+                } else if (obj.tags.type == "backend") {
+                  backend.push(obj.name);
+                }
+              }
+            );
+
+            function getPagesObj(
+              instance: CoreTemplateStudio,
+              projType: string,
```

Comment:
These parameters (instance, projType) will hide those declared above. Is this intentional?